homeassistant/components/mqtt/config_flow.py

```diff
@@ -648,6 +670,9 @@ def validate_sensor_platform_config(
     ):
         errors[CONF_UNIT_OF_MEASUREMENT] = "invalid_uom_for_state_class"
 
+    if config.get(CONF_ENTITY_CATEGORY) == EntityCategory.CONFIG:
+        errors[CONF_ENTITY_CATEGORY] = "sensor_entity_category_must_not_be_config"
```

Comment:
Instead of adding this here and in binary sensor validation, why don't we make a custom select selector which doesn't have `EntityCategory.CONFIG`?

Reply:
Step moved. Because there is always an option for entity platform config step, it is no longer conditional. Because of this some tests needed an update.